src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -96,36 +96,84 @@ namespace AppInstaller::CLI::Workflow
             return {};
         }
 
-        void OutputValueSet(OutputStream& out, const ValueSet& valueSet, size_t indent)
+        void OutputPropertyValue(OutputStream& out, const IPropertyValue property)
+        {
+            switch (property.Type())
+            {
+            case PropertyType::String:
+                out << ' ' << Utility::ConvertToUTF8(property.GetString()) << '\n';
+                break;
+            case PropertyType::Boolean:
+                out << ' ' << (property.GetBoolean() ? "true" : "false") << '\n';
+                break;
+            case PropertyType::Int64:
+                out << ' ' << property.GetInt64() << '\n';
+                break;
+            default:
+                out << " [Debug:PropertyType="_liv << property.Type() << "]\n"_liv;
+                break;
+            }
+        }
+
+        void OutputValueSet(OutputStream& out, const ValueSet& valueSet, size_t indent, bool isArray)
         {
             Utility::LocIndString indentString{ std::string(indent, ' ') };
+            bool processedFirstElement = false;
 
             for (const auto& value : valueSet)
             {
-                out << indentString << Utility::ConvertToUTF8(value.Key()) << ':';
+                if (isArray && !processedFirstElement)

```

Comment:
I would check for `treatAsArray` first and defer to a separate `OuputValueSetAsArray` function to keep things less complicated in this function.

PS: Thought experiment:
```YAML
objectButArray:
  0: Item0
  1: Item1
  treatAsArray: true
```